pandas/core/strings.py

```diff
@@ -456,9 +456,13 @@ def str_replace(arr, pat, repl, n=-1, case=None, flags=0, regex=True):
     flags : int, default 0 (no flags)
         - re module flags, e.g. re.IGNORECASE
         - Cannot be set if `pat` is a compiled regex
-    regex : bool, default True
+    regex : boolean, default None
```

Comment:
default is still True

Reply:
so, you mean we now dont raise a warning anymore when char is a special single character while user doesn't explicitly specify `regex`?
if set default to `True`, and i use @TomAugspurger example:
```python
pd.Series(['aa']).str.replace('.', 'b')
```
we will get:
1. `default` or explicitly `regex=True`, output is `bb`, no warning
2. `regex=False`, output is `aa`, no warning.
is it correct? @WillAyd @TomAugspurger @jreback